src/AddPlayerPacket.php

```diff
@@ -161,7 +162,7 @@ protected function encodePayload(PacketSerializer $out) : void{
 		if($out->getProtocolId() >= ProtocolInfo::PROTOCOL_1_18_30){
 			$out->putVarInt($this->gameMode);
 		}
-		$out->putEntityMetadata($this->metadata);
+		$out->putEntityMetadata(EntityMetadataProperties::convertProps($out->getProtocolId(), $this->metadata));
```

Comment:
Maybe you should put this in the function itself, as it'll reduce the usage to 1